### PR TITLE
Add maven dependency version reports

### PR DESCRIPTION
**Pull request by cfkoehler**
Not sure if this is something we want but thought it was helpful when looking at what dependency could be updated.

By running `mvn site` the reports are generated. The reports can also be generated by using the mvn versions target `mvn versions:display-dependency-updates`

The output with site adds the reports and looks like this:
image

### CODE REVIEW

**Comment by ldhardy** [comment]
This is handy.

**Review by ldhardy — DISMISSED** [review]

**Review by drivenflywheel — DISMISSED** [review]
I like this.  Should we add a note/instruction to our of the repo README files?

**Comment by cfkoehler** [comment]
> I like this. Should we add a note/instruction to our of the repo README files?

That is a good idea. Will do
- [ ] add note in the readme

File: DEVELOPING.md [hunk 1]
```diff
@@ -483,6 +483,16 @@ debugger attaches, and you can then step through the code.
 mvn clean test -Dtest=ServerCommandTest -Dmaven.surefire.debug="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=8000 -Xnoagent -Djava.compiler=NONE"
 ```
 
+### View Dependency Reports
+
+To generate a report to display plugins, properties, dependency that have updates available run the following maven commands.
```
Comment (jpdahlke) [inline DEVELOPING.md:488]:
```suggestion
To generate a report to display plugins, properties, dependencies that have updates available run the following maven commands.
```

**Review by jpdahlke — APPROVED** [review]

**Review by dev-mlb — APPROVED** [review]

**Pull request merged**